### PR TITLE
[COM-243] Spring provides authorization based on Vendia roles

### PR DESCRIPTION
**Pull request by dvillavicencio**
This PR implements a simple reactive authorization manager that checks the roles of a given user using data from Vendia. Some things to note is that this is gonna be used in the Spring Security configuration file `/config/SecurityConfiguration.java` and it can be used to add authorization to many paths in our project. The current usage I could think of, was for `POST` requests under the `/api/appointments/**` route, but this can be subject to change depending on our application needs.

### CODE REVIEW

**Review by arielcamargo — APPROVED** [review]

File: src/main/java/com/compilercharisma/chameleonbusinessstudio/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -13,17 +15,13 @@ public class SecurityConfiguration {
...
-                .authorizeExchange()
-                .pathMatchers("/api/appointments/**")
-                .permitAll()
-                .and()
                 .oauth2Login()
                 .and()
-                .logout()
-                .and()
                 .authorizeExchange()
+                .pathMatchers(HttpMethod.POST, "/api/appointments/**").access(userAuthorizationManager)
```
Comment (rramsdany) [inline src/main/java/com/compilercharisma/chameleonbusinessstudio/config/SecurityConfiguration.java:23]: This is dope!

**Review by rramsdany — APPROVED** [review]

**Review by Matt-Crow — APPROVED** [review]
Nice!

**Comment by rramsdany** [comment]
Is this PR ready to be merged?

**Comment by dvillavicencio** [comment]
My bad, I forgot to merge

**Pull request merged**